IL: Update LTRF

patwey opened the pull request:
## Context & Motivation

<!-- Required: Why is this change needed? Link to issue, describe the problem, or explain the goal -->

- Fixes: https://linear.app/myfriendben/issue/MFB-190/il-add-local-transit-reduced-fare
- Related PR: https://github.com/MyFriendBen/benefits-api/pull/1128

## Changes Made

<!-- Required: What specifically changed? Be concrete and specific -->

- Add to list of eligible counties
- Add special minimum age for Rock Island county

<img width="1053" height="713" alt="Screenshot 2025-09-08 at 2 45 43 PM" src="https://github.com/user-attachments/assets/9ae63ddb-247e-4461-9180-5a7746e7f205" />

<img width="1300" height="150" alt="Screenshot 2025-09-08 at 2 45 27 PM" src="https://github.com/user-attachments/assets/6fb7a14c-a78e-4ced-84be-53d5fddbcbc8" />

## Testing

<!-- Steps needed to test this PR locally -->

- Manual testing steps:
    * Create a screen for a 60-64 year old in Rock Island county with a high income
    * Verify that LTRF is included in the eligibility results

## Deployment

<!-- Steps needed AFTER merging to get this live -->

- Admin updates needed: Add `Local Transit Reduced Fare` to the `Benefit Access` program's `excludes_programs` set

jmejia commented on programs/programs/il/transit_reduced_fare/calculator.py:
```diff
@@ -1,11 +1,27 @@
+        "rock island",
+        "sangamon",
+        "will",
+    ]
```
I think I have a slight preference for consistency to make it easy for devs to bounce between white labels. That said, I'm not opposed to the `lower` approach but would probably want to make that the norm across white labels if we like it better.

patwey replied:
Totally makes sense, I'll update it to use title case 👍🏻

patwey commented:
@jmejia just wanted to highlight that I discovered an edge case in my testing and fixed it with this commit: https://github.com/MyFriendBen/benefits-api/pull/1154/commits/a90e4daabcd01dfc8fbaa1cff44d8afe719477f5

TL;DR there isn't actually an income limit for LTRF in the policy. We prefer to show BAP if a HH is eligible for both, because it's the better benefit, but there's no income eligibility requirement. I've removed that check from the code, and we'll use the `excludes_programs` functionality to hide LTRF when there's BAP eligibility (this also fixes a immigration filter related edge case).
